### PR TITLE
Fixes #30671 - Send UI Notification when Bastion table cannot load it…

### PR DESCRIPTION
**Pull request by jturel**
…s resource

Looks like a very old regression. The way we were handling errors from the table resource was confusing and seemingly wrong! A simple way to test is to go to the content hosts list and search 'errata_status = foo'. Previously there'd be an error about an unhandled promise in the console, but now it's cleaned up and sends a UI notification. Since this touches a critical area, I recommend doing any action that would refresh the table (such as paging, searching) and make sure everything looks good and no errors in console.

### CODE REVIEW

File: engines/bastion/app/assets/javascripts/bastion/components/nutupane.factory.js [hunk 1]
```diff
@@ -155,14 +150,14 @@ angular.module('Bastion.components').factory('Nutupane',
                     table.working = false;
                     table.refreshing = false;
                     table.initialLoad = false;
-                });
+                }).catch(function(response) {
+                    table.working = false;
+                    table.refreshing = false;
 
-                if (resourceCall && resourceCall.$promise && resourceCall.$promise.catch) {
```
Comment (jturel) [inline engines/bastion/app/assets/javascripts/bastion/components/nutupane.factory.js:160]: I tried to send the Notification in this block but the 'unhandled promise' error was still being thrown in the browser console. use $promise.then.catch is the correct way to handle this, so I adjusted the code and fixed related test failures.

**Comment by jturel** [comment]
@jeremylenz updated!

**Comment by jeremylenz** [comment]
@jturel Changes look great, no more console error :)

Looks like some Angular tests still need adjusting, but ACK pending that.

**Comment by jturel** [comment]
test fix pushed. APJ?

**Comment by jeremylenz** [comment]
I was slower than J.  ACK (remember to squash ;)

**Pull request merged**